display/ssd1306.h

```diff
@@ -445,70 +473,124 @@ class StandardDisplayController : public DisplayControllerBase<Width, col_t>, Sa
     }
   }
 
+  void SB_On2() override {
+    if (IMG_out) {
+      ShowFileWithSoundLength(&IMG_out, font_config.ProffieOSOutImageDuration);
+    }
+  }
+
   void usb_connected() override {
     if (EscapeIdleIfNeeded() && current_effect_ == &IMG_idle) {
       // We are idle-looping, and usb is connected. Time to stop.
-      SetMessage("usb");
+      SetMessage("    usb\nconnected");
       SetScreenNow(SCREEN_MESSAGE);
     }
   }
 
-  void SB_Effect(EffectType effect, float location) override {
+  void ShowFileWithSoundLength(Effect* e, float duration) {
+    float dur = (duration == 0) ? SaberBase::sound_length * 1000 : duration;
+    ShowFile(e, round(dur));
+  }
+
+ void SB_Effect(EffectType effect, float location) override {
```

Comment:
Might be easier to just move everything into SB_effect2

Reply:
I honestly don't quite get what that is/does/difference. 
I will remove All things then from SB_EFffect into SB_Effect2 ?

Reply:
moved. SB_Off2 also.

Reply:
SB_****2 was added specifically so that SB_**** can run *first* and start any sounds that needs to play.
Afterwards we call SB_****2, which will be able to know what sound was started, how long it is etc.

Reply:
Perfect, thanks.